numpy/lib/function_base.py

```diff
@@ -4368,6 +4368,19 @@ def delete(arr, obj, axis=None):
             return new
 
     if isinstance(obj, (int, integer)) and not isinstance(obj, bool):
+        single_value = True
+    else:
+        single_value = False
+        _obj = obj
+        obj = np.asarray(obj)
+        if obj.size == 0 and not isinstance(_obj, np.ndarray):
+            obj = obj.astype(intp)
+
+        elif obj.shape == (1,):
```

Comment:
You can hit a lot more paths here with 
```suggestion
        elif obj.size== 1:
```